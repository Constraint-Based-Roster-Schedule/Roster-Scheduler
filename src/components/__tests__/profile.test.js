import { render, screen } from "@testing-library/react";
import { getByTestId } from "@testing-library/react";
import DoctorProfile from "../login";
import ReactDom from "react-dom";
import { Button } from "react-bootstrap";
import renderer from "react-test-renderer";
it("renders without crashing testing", () => {
  const div = document.createElement("div");
  ReactDom.render(<Button></Button>, div);
});

test("doctor profile testing", () => {
  render(<DoctorProfile />);
  const nameLabel = screen.getByText(/Type/i);
  const nameLabe2 = screen.getByText(/Password/i);
  const nameLabe3 = screen.getByText(/Email/i);
  const nameLabe4 = screen.getByText(/Submit/i);
  const nameLabe5 = screen.getByText(/Welcome/i);
  expect(nameLabel).toBeInTheDocument();
  // expect(getByTestId("doctor-profile-title")).toHaveTextContent("Doctor Profile");
});
it("matches doctor profile snapshot", () => {
  const tree = renderer.create(<DoctorProfile />).toJSON();
  expect(tree).toMatchSnapshot();
});
