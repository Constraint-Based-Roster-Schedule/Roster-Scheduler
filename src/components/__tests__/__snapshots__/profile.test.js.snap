// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches doctor profile snapshot 1`] = `
<div
  className="main-login"
>
  <br />
  <div
    className="login-container"
  >
    <div
      className="left-side"
    >
      <div
        className="img-class"
      >
        <img
          alt=""
          src="logo.png"
        />
      </div>
      <form
        onSubmit={[Function]}
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Enter your email.."
          required={true}
          type="email"
          value=""
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          required={true}
          type="password"
          value=""
        />
        <label
          for="type"
        >
          Type
        </label>
        <select
          id="type"
          name="type"
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            disabled={true}
            hidden={true}
            selected={true}
            value=""
          >
             
            Select
             
          </option>
           
          <option
            id="type"
            name="1"
            value="1"
          >
            Doctor
          </option>
          <option
            id="type"
            name="3"
            value="3"
          >
            Admin
          </option>
          <option
            id="type"
            name="2"
            value="2"
          >
            Consultant
          </option>
        </select>
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
      <div
        className="footer"
      >
        <h4 />
      </div>
    </div>
    <div
      className="right-side"
    >
      <div
        className="welcome-note"
      >
        <h3>
          ........Welcome Back........
        </h3>
      </div>
      <div
        className="welcome-img"
      >
        <img
          alt=""
          id="wel-img-id"
          src="doctor.png"
        />
      </div>
    </div>
  </div>
</div>
`;
